src/modules/notifSettings/EditNotificationSetting.form.tsx

```diff
@@ -0,0 +1,153 @@
+import { FC } from 'react';
+
+import * as Yup from 'yup';
+
+import { FormikProvider, useFormik, useFormikContext } from 'formik';
+
+import { getMinCharLimitMessage, getRequiredFieldMessage } from 'common';
+
+import { ChannelSetting } from 'modules/channelDashboard/ChannelDashboard.types';
+
+type EditNotificationSettingsFormProviderProps = {
+  children: React.ReactNode;
+  initialValue: ChannelSetting;
+  onSubmit: (values: NotificationSettingFormValues) => void;
+};
+
+export const getFormInitialValus = (initialValue: ChannelSetting): NotificationSettingFormValues => {
+  return {
+    settingName: initialValue.description,
+    isDefault:
+      initialValue.type === 1
+        ? typeof initialValue.default === 'boolean'
+          ? initialValue.default
+          : true
+        : initialValue.enabled!,
+    enableRange: initialValue.type !== 1 ? true : false,
+    rangelowerlimit: initialValue.lowerLimit ? initialValue.lowerLimit : 0,
+    rangeupperlimit: initialValue.upperLimit ? initialValue.upperLimit : 0,
+    enableMultiRange: initialValue.type === 3 ? true : false,
+    defaultValue: typeof initialValue.default === 'number' ? initialValue.default : 0,
+    multirangelowerlimit: typeof initialValue.default === 'object' ? initialValue.default.lower : 0,
+    multirangeupperlimit: typeof initialValue.default === 'object' ? initialValue.default.upper : 0,
+    sliderStepValue: initialValue.ticker ? initialValue.ticker : 0,
+  };
+};
+
+export const NotificationSettingsSchema = Yup.object().shape({
+  settingName: Yup.string().required(getRequiredFieldMessage('Setting Name')),
+  isDefault: Yup.boolean(),
+  enableRange: Yup.boolean(),
+
+  rangelowerlimit: Yup.number().when('enableRange', {
+    is: true,
+    then: () => Yup.number().min(1, getMinCharLimitMessage(1)).required(getRequiredFieldMessage('Range')),
+    otherwise: () => Yup.number(),
+  }),
+
+  rangeupperlimit: Yup.number().when('enableRange', {
+    is: true,
+    then: () =>
+      Yup.number()
+        .min(Yup.ref('rangelowerlimit'), getMinCharLimitMessage('lower limit'))
+        .required(getRequiredFieldMessage('Range')),
+    otherwise: () => Yup.number(),
```

Comment:
We also need to have a proper error message function for numeric values. eg. Upper limit must be a number. 

Do ask gpt for better suggestions.